src/api/sendgrid.ts

```diff
@@ -0,0 +1,53 @@
+import mail, { MailDataRequired } from '@sendgrid/mail'
+import apiEnv from './apiEnv';
+
+export interface Personalizations {
+    to: [{
+        email: string,
+        name: string
+    }];
+    dynamic_template_data?: Record<string, string>;
+}
+
+mail.setApiKey(apiEnv.SENDGRID_API_KEY);
+
+const sendEmail = async (personalizations: Personalizations[], template_id: string) => {
+
+    console.log("SendGrid Mail Executed");
```

Comment:
it's not executed yet. print `Sending mail via SendGrid: ${JSON.stringify(personalizations, null, 2)}` 

also, move this line right above calling mail.send()